LeaderboardBackend/Controllers/RunsController.cs

```diff
@@ -34,30 +38,32 @@ public async Task<ActionResult<RunViewModel>> GetRun(Guid id)
     }
 
     [Authorize]
-    [HttpPost("runs/create")]
-    [SwaggerOperation("Creates a new Run.", OperationId = "createRun")]
+    [HttpPost("/category/{id:long}/runs/create")]
+    [SwaggerOperation("Creates a new Run for a Category with ID `id`. This request is restricted to confirmed Users and Administrators.", OperationId = "createRun")]
     [SwaggerResponse(201)]
-    [SwaggerResponse(401)]
-    [SwaggerResponse(403)]
+    [SwaggerResponse(401, "The client is not logged in.", typeof(ProblemDetails))]
+    [SwaggerResponse(403, "The requesting User is unauthorized to create Runs.", typeof(ProblemDetails))]
+    [SwaggerResponse(404, "The Category with ID `id` could not be found.", typeof(ProblemDetails))]
     [SwaggerResponse(422, Type = typeof(ValidationProblemDetails))]
-    public async Task<ActionResult> CreateRun([FromBody] CreateRunRequest request)
+    public async Task<ActionResult<RunViewModel>> CreateRun([FromRoute] long id, [FromBody, SwaggerRequestBody(Required = true)] JsonDocument request)
     {
-        // FIXME: Should return Task<ActionResult<Run>>! - Ero
-        // NOTE: Return NotFound for anything in here? - Ero
-
         GetUserResult res = await userService.GetUserFromClaims(HttpContext.User);
 
         if (res.TryPickT0(out User user, out OneOf<BadCredentials, UserNotFound> _))
         {
-            Run run = new()
+            CreateRunRequest? deserialised = JsonSerializer.Deserialize<CreateRunRequest>(request, jsonOptions.Value);
```

Comment:
This bit here fails. I get an error saying the non-derived properties (PlayedOn and Info) aren't deserialized.